src/Containers/Microsoft.NET.Build.Containers/SourceImageReference.cs

```diff
@@ -4,24 +4,43 @@
 namespace Microsoft.NET.Build.Containers;
 
 /// <summary>
-/// Represents a reference to a Docker image. A reference is made of a registry, a repository (aka the image name) and a tag.
+/// Represents a reference to a Docker image. A reference is made of a registry, a repository (aka the image name) and a tag or digest.
 /// </summary>
-internal readonly record struct SourceImageReference(Registry? Registry, string Repository, string Tag)
+internal readonly record struct SourceImageReference(Registry? Registry, string Repository, string Tag, string? Digest)
```

Comment:
Yes, good idea. I like it.

One question: How do we differentiate between the constructors since both `Tag` and `Digest` are strings? 🤔

Reply:
It might be easier to make static Factory functions in this case:

```
SourceImageReference.FromTag(....)
SourceImageReference.FromDigest(...)
```

Reply:
Good idea.

Should I keep the current constructor public, or make it private and refactor the other uses to `FromTag`?

Not sure how many they are.

Reply:
Let's go ahead and make them private + update usages if that's ok - that will prevent invalid construction.

Reply:
Note that the user may specify an image reference that includes both a tag and a digest.

Probably the SDK doesn't care much about the tag if a digest is specified.

I don't know what you gain by having separate methods since the SDK must handle both the case where a digest is specified and the case where (only) the tag is specified.

Reply:
Given what @tmds said, mabye the logic here can be simplified.

This is a totally legit reference:

```
mcr.microsoft.com/dotnet/aspnet:8.0@sha256:6cec36412a215aad2a033cfe259890482be0a1dcb680e81fccc393b2d4069455
```

So maybe if the digest is not null or empty, we just append it to the reference?

Reply:
I changed it to this: https://github.com/BeyondEvil/dotnet-sdk/blob/beyondevil/support-image-digest/src/Containers/Microsoft.NET.Build.Containers/SourceImageReference.cs

But I'm not sure if that is idiomatic to .NET?

Reply:
ping @baronfel  ☝️ 😊

Reply:
Oh this was tricky :D The name `RepositoryAndTag` made me think that it would be invalid if the `Tag` was empty, but it turns out that all of this still works. That's fine by me.